src/Components/CommentCard.jsx

```diff
@@ -3,44 +3,44 @@ import { useEffect } from "react";
 import { useState } from "react";
 import axios from "axios";
 import { FaTimes } from "react-icons/fa";
+import { fetchReviewComments, postReviewComment } from "../Components/Api";
+import "../StyleSheets/CommentCard.css";
 
 const CommentCard = ({ review_id }) => {
   const [comments, setComments] = useState([]);
   const id = review_id;
   const [err, setErr] = useState(null);
+  const [value, setValue] = useState("");
 
   useEffect(() => {
-    axios
-      .get(`https://games-nc.herokuapp.com/api/reviews/${id}/comments`)
-      .then((res) => {
-        setComments(res.data.comments);
-      });
-  }, [id]);
+    fetchReviewComments(id).then((data) => {
+      setComments(data.comments);
+    });
+  }, [comments.length, id]);
 
   const handleSubmit = (e) => {
     e.preventDefault();
-    let comment = e.target.firstElementChild.value;
+    let body = e.target.firstElementChild.value;
+    let username = "tickle122";
     let commentObj = {
-      body: comment,
+      body: "uploading comment",
       comment_id: Math.random(),
-      author: "tickle122",
     };
     setComments([commentObj, ...comments]);
-    document.getElementById("newcomment").value = "";
-
-    axios
-      .post(`https://games-nc.herokuapp.com/api/reviews/${id}/comments`, {
-        username: "tickle122",
-        body: comment,
-      })
-      .catch((err) => {
-        setErr("oops, We could not post that comment. Please try again later.");
-      });
+    setValue("");
+    postReviewComment(id, username, body).catch((err) => {
+      setErr("oops, We could not post that comment. Please try again later.");
+    });
   };
 
   const handleClick = (index, comment_id) => {
     setComments((currcomments) => {
-      currcomments.splice(index, 1);
+      currcomments.splice(
+        index,
+        1,
+        { body: "deleting comment", comment_id: Math.random() },
+        { body: "", comment_id: Math.random() }
```

Comment:
here as well, we don't want to be setting comment_id ourselves ever on the front end.

When a comment is added, you will get a return value from the api which is the comment object.  this object contains the comment_id, and it is that value that you should be using throughout.